packages/connection-storage/src/connection-storage.ts

```diff
@@ -326,34 +326,26 @@ export class ConnectionStorage {
       // where we use storage-mixin. storage-mixin uses this prop
       // to map keytar credentials to the stored connection.
 
-      // While testing, we don't use keychain to store secrets
-      if (process.env.COMPASS_E2E_DISABLE_KEYCHAIN_USAGE === 'true') {
-        await this.userData.write(connectionInfo.id, {
-          connectionInfo,
-          _id: connectionInfo.id,
-        });
-      } else {
-        const { secrets, connectionInfo: connectionInfoWithoutSecrets } =
-          extractSecrets(connectionInfo);
-
-        let connectionSecrets: string | undefined = undefined;
-        try {
-          connectionSecrets = this.encryptSecrets(secrets);
-        } catch (e) {
-          log.error(
-            mongoLogId(1_001_000_202),
-            'Connection Storage',
-            'Failed to encrypt secrets',
-            { message: (e as Error).message }
-          );
-        }
-        await this.userData.write(connectionInfo.id, {
-          _id: connectionInfo.id,
-          connectionInfo: connectionInfoWithoutSecrets,
-          connectionSecrets,
-          version: this.version,
-        });
+      const { secrets, connectionInfo: connectionInfoWithoutSecrets } =
+        extractSecrets(connectionInfo);
+
+      let connectionSecrets: string | undefined = undefined;
+      try {
+        connectionSecrets = this.encryptSecrets(secrets);
+      } catch (e) {
+        log.error(
```

Comment:
I realise that this is existing behaviour, but shouldn't this re-throw the error rather than log and swallowing it? If encryptSecrets fails we'll write unencrypted ones? Is that behaviour we always want?

Reply:
we want users to be able to save the connections even if secrets can not be saved. for some linux distros, if keychain is not available and instead of throwing, we still save the connection.

Reply:
there's already a similar issue reported https://github.com/mongodb-js/compass/issues/1911

Reply:
I agree, but my concern is just doing that for any error and totally transparently.